did-docs.html

```diff
@@ -114,6 +112,23 @@ <h2>CRDTs</h2>
 
 </section>
 
+<section>
+    <h2><code>initial-state</code></h2>
+    <p>
+        Sometimes it is useful (or even required) to communicate a doc-capable peer DID value (one with <code>numalgo</code> != 0) and its genesis or evolved DID doc as a single combined value, instead of sharing the DID as one piece of data, and its DID doc and deltas as another. This situation commonly occurs in connection protocols that are built on <a target="_blank" href="https://identity.foundation/didcomm-messaging/docs/spec/">DIDComm messaging</a>, for example: <var>A</var> wants to share with <var>B</var> the new pairwise DID they've allocated, plus the DID's full DID doc, and DIDComm requires all information about the sender to be passed in a single <code>from</code> field.
+    </p>
+    <p>
+        The way to do this is to use the <code>initial-state</code> query parameter of a peer DID URI. All implementations of peer DIDs are required to recognize the meaning of this query parameter, though only implementations that support deltas (layer 2 or 3) need to process it. The parameter value can be extracted from a peer DID URI with the following regex:
+    </p>
+    <pre class="example nohighlight" title="Regex to extract initial-state" id="initial-state-regex">
+^did:peer:[^0].+[?](.*&)?initial-state=([^&]*)(?:&|$)
+    </pre>
+    <p>
+        This places the value of the <code>initial-state</code> parameter in capture group 1.
+    </p>
+    <p>To interpret the parameter value, it must first be base64url decoded. The result MUST be a JSON array containing zero or more deltas. An empty parameter value is also allowed, and is considered equivalent to an empty JSON array.</p>
```

Comment:
I'm a bit confused about what you're describing with the JSON array containing zero or more deltas. Is the recipient expected to rebuild the did document from the various deltas provided in the initial-value parameters?

In sidetree just the did document is provided in base64url encoding, but there's no requirements about the canonicalization of the data in the document unlike what would occur with the deltas to get a proper output from the CRDT by the reciever. So, my interpretation is the value of this parameter is slightly different than what's in sidetree, but the end goal is the same (pass the information in a did document as a query parameter)

Reply:
The peer DID spec differentiates between the **stored variant** of a DID doc and the **resolved variant**. The resolved variant is a canonical DID doc. The stored variant is a set of deltas from which a resolved variant can be read. The deltas are what must be transmitted in initial state, not the rendered doc.